src/deploy/functions/args.ts

```diff
@@ -0,0 +1,78 @@
+// These types should proably be in a root deploy.ts, but we can only boil the ocean one bit at a time.
+
+import { ReadStream } from "fs";
+import * as gcf from "../../gcp/cloudfunctions";
+import * as deploymentPlanner from "./deploymentPlanner";
+
+// Payload holds the output types of what we're building.
+export interface Payload {
+  functions?: {
+    byRegion: deploymentPlanner.RegionMap;
+    triggers: deploymentPlanner.CloudFunctionTrigger[];
+  };
+}
+
+// Options come from command-line options and stored config values
+// TODO: actually define all of this stuff in command.ts and import it from there.
+export interface Options {
+  cwd: string;
+  configPath: string;
+  // Why is there options.project and context.projectId?
+  project: string;
+
+  only: string;
+  // defined in /config.js
+  config: {
+    // Note: it might be worth defining overloads for config values we use in
+    // deploy/functions.
+    get(key: string, defaultValue?: any): any;
+    set(key: string, value: any): void;
+    has(key: string): boolean;
+    path(pathName: string): string;
+
+    // I/O methods: these methods work with JSON objects.
+    // WARNING: they all use synchronous I/O
+    readProjectFile(file: string): unknown;
+    writeProjectFile(path: string, content: unknown): void;
+    askWriteProjectFile(path: string, content: unknown): void;
+
+    projectDir: string;
+  };
+  filteredTargets: string[];
+  nonInteractive: boolean;
+  force: boolean;
+}
+
+export interface FunctionsSource {
```

Comment:
Hmm... interesting point. In my work-in-progress branch I have `backend.existingBackend(context)`. Maybe I could do `prepareFunctionsUpload.functionsSource(context)`? I don't know that I want to have `functionsSource: prepareFunctionsUpload.FunctionsSource` because it'll create a weird dependency cycle.

How do you feel? (also, as an aside, I hate that we have file names named after a command, yet they act as a namespace for multiple things. Go-style namespacing ftw IMO)